test/parallel/test-pipe-writev.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const net = require('net');
+
+if (common.isWindows) {
+  common.skip('Unix-specific test');
+  return;
+}
+
+common.refreshTmpDir();
+
+const server = net.createServer((connection) => {
+  connection.on('error', (err) => {
+    throw err;
+  });
+
+  const writev = connection._writev.bind(connection);
+  connection._writev = common.mustCall(writev);
+
+  connection.cork();
+  connection.write('pi');
+  setTimeout(() => {
+    connection.write('ng');
+    connection.end();
+  }, 200);
```

Comment:
Hmm... maybe it is even better to remove the `setTimeout` completely since there's a `common.mustCall` on `connection._writev` and even 200 ms timeout doesn't guarantee that a client connection will emit two `data` events.